pkg/utils/clusters/shoot_client.go

```diff
@@ -56,7 +56,7 @@ func NewShootClientFromTarget(ctx context.Context, gardenTarget *v1alpha1.Target
 	}
 
 	targetConfig := &targettypes.KubernetesClusterTargetConfig{}
-	err = json.Unmarshal([]byte(resolvedTarget.Content), targetConfig)
+	err = yaml.Unmarshal([]byte(resolvedTarget.Content), targetConfig)
```

Comment:
I guess this is fine, since the yaml parser should also be able to parse json (?).
However, I believe we need to also fix the secret resolving in `controller-utils/pkg/landscaper/references.go`. The function `ResolveSecretReference` should return the same format for when a key is specivied vs. there is no key specified. What do you think?